app/models/affair.rb

```diff
@@ -154,6 +154,16 @@ def affairable_types
       @affairable_types ||= %w[SaleGap PurchaseGap Sale Purchase Payslip IncomingPayment OutgoingPayment Regularization DebtTransfer].freeze
     end
 
+    def affairable_types_entity_foreign_keys
```

Comment:
We should forego this and use polymorphism instead (I think we already do with the name `third`.)